src/main/java/org/folio/processing/matching/loader/query/LoadQueryBuilder.java

```diff
@@ -54,6 +58,13 @@ public static LoadQuery build(Value value, MatchDetail matchDetail) {
             QueryHolder additionalQuery = new QueryHolder(StringValue.of(additionalField.getValue()), matchDetail.getMatchCriterion())
               .replaceFieldReference(additionalFieldName, true);
             mainQuery.applyAdditionalCondition(additionalQuery);
+            // TODO provide all the requirements for MODDATAIMP-592 and refactor code block below
+            if(matchDetail.getIncomingRecordType() == EntityType.MARC_BIBLIOGRAPHIC && matchDetail.getExistingRecordType() == EntityType.INSTANCE &&
+                matchDetail.getMatchCriterion() == MatchDetail.MatchCriterion.EXACTLY_MATCHES && fieldPath.equals(IDENTIFIER_TYPE_VALUE) &&
+                additionalField.getLabel().equals(IDENTIFIER_TYPE_ID)) {
```

Comment:
@Abdulkhakimov  Could you please extract this huge check to a separate method? For example:  "checkIfIdentifierTypeExists"

Reply:
Done